docs/user-guide/route-modules.md

```diff
@@ -1,3 +1,369 @@
-# Route Modules
+# Modules
 
-<img src="../assets/images/Under-Construction-Sign.png" alt="Under Construction" width="350" />
+
+ngrok's Cloud Edge [Modules](https://ngrok.com/docs/cloud-edge/modules/) allow you to configure features like compression, IP Restrictions, OAuth, adding/removing headers, and more.
+
+<!-- TOC depthfrom:2 -->
+
+- [Design](#design)
+    - [Reusable](#reusable)
+    - [Composable](#composable)
+- [Supported Modules](#supported-modules)
+    - [Circuit Breaker](#circuit-breaker)
+    - [Compression](#compression)
+        - [Enabled](#enabled)
+        - [Disabled](#disabled)
+    - [Headers](#headers)
+        - [Request](#request)
+        - [Response](#response)
+    - [IP Restrictions](#ip-restrictions)
+    - [OAuth](#oauth)
+        - [Ngrok Managed OAuth Application](#ngrok-managed-oauth-application)
+            - [Google](#google)
+        - [User Managed OAuth Application](#user-managed-oauth-application)
+            - [Google](#google)
+    - [OpenID Connect OIDC](#openid-connect-oidc)
+    - [SAML](#saml)
+    - [TLS Termination](#tls-termination)
+    - [Webhook Verification](#webhook-verification)
+
+<!-- /TOC -->
+
+## Design
+
+### Reusable
+
+`NgrokModuleSet`s are designed to be reusable. This allows you to define a set of modules and their configuration once and apply it to multiple Ingresses. Ex:
+
+```yaml
+---
+kind: NgrokModuleSet
+apiVersion: ingress.k8s.ngrok.com/v1alpha1
+metadata:
+  name: module-set-1
+modules:
+  compression:
+    enabled: true
+  tlsTermination:
+    minVersion: "1.2"
+  headers:
+    request:
+      add:
+        a-request-header: "my-custom-value"
+        another-request-header: "my-other-custom-value"
+      remove:
+      - "x-remove-at-edge"
+    response:
+      add:
+        a-response-header: "a-response-value"
+---
+kind: Ingress
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: example-ingress
+  annotations:
+    k8s.ngrok.com/modules: module-set-1
+---
+kind: Ingress
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: example-ingress-2
+  annotations:
+    k8s.ngrok.com/modules: module-set-1
+```
+
+In this example, the `compression`, `tlsTermination`, and `headers` modules are applied to both Ingresses and the same configuration is used for both. If you change the configuration of the `NgrokModuleSet`, the change will be applied to all Ingresses that use it.
+
+
+### Composable
+
+`NgrokModuleSet`s are designed to be composable. If multiple `NgrokModuleSet`s are applied to an Ingress and a module is configured in more than one, the last one wins. Ex:
```

Comment:
The other neat part is technically you can use rbac to let someone manage these resources for important things like oauth/tls/etc and prevent others from modifying it while letting them still self serve ingress. Do you think thats worth noting in this doc or maybe we should have an general rbac doc eventually like https://kubernetes.github.io/ingress-nginx/deploy/rbac/ ?

Reply:
Good call. I'll add a separate call out for that